### PR TITLE
Upgrade snappy to 1.1.7

### LINKED ISSUE #521: Update Snappy to 1.1.7
See https://github.com/Level/rocksdb/pull/79

### PR DESCRIPTION
**Pull request by filoozom**
Fixes #521

### CODE REVIEW

**Comment by ralphtheninja** [comment]
> Lost one more line for some reason, strange. Shouldn't we move this to a Git Submodule now that I think about it?

It's probably wise to do so now yes. But might not be completely straight forward. Needs to get some work to get paths right etc (e.g. `snappy.gyp` shouldn't use version numbers in path)

**Comment by filoozom** [comment]
This is actually great because we don't need to push the whole C++ codebase to `npm` for users that can just make use of `prebuild` packages.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "deps/snappy/snappy"]
+	path = deps/snappy/snappy
+	url = https://github.com/google/snappy.git
```
Comment (peakji) [inline .gitmodules]: Can we specify a commit tag here to lock the version?
I remember that git submodule will always update to the latest commit of a branch. So future changes in snappy might break our code?

Comment (filoozom) [reply .gitmodules]: It should already be fixed to the `1.1.7` tag. Can you try by checking out this commit?

**Comment by filoozom** [comment]
I'd like to add something like this to `package.json`:
```json
{
  "scripts": {
    "dev": "git submodule update --init && node-gyp rebuild",
    "install": "prebuild-install || npm run dev && git submodule deinit .",
  }
}
```

(I chose `dev` instead of `build` because if I add a non-empty `build` script it automatically runs `prebuild` for some reason. Maybe this can be fixed or there's a better name?)

This would be nice for users because they wouldn't end up with a few megabytes of useless C++ dependencies, but would require developers and CI to download all git submodules twice. Any ideas?

Except for that, this PR is ready. Maybe I should work on that on another PR actually.

**Comment by ralphtheninja** [comment]
Actually, I don't think we should do the `git submodule update`-dance during the install phase. This will cause problems when people are installing it from npm. The code should still be published together with the package, so everything is in the tarball from npm.

**Comment by filoozom** [comment]
That's a good point. Wouldn't work if installing on a system without `git`. Guess there's actually no way to make the package smaller after an install then. 🙁

At least this should make it easier to update upstream dependencies (`cd deps/snappy/snappy; git checkout $version`, commit the thing and there you go).

**Comment by ralphtheninja** [comment]
> At least this should make it easier to update upstream dependencies (`cd deps/snappy/snappy; git checkout $version`, commit the thing and there you go).

Yes, it's a huge improvement when it comes to maintenance :) We should do this with all code that's vendored, e.g. leveldb and rocksdb as well.

**Comment by filoozom** [comment]
@ralphtheninja I'm already on it. 😉 Just making sure that all versions are exactly the same and that I'm not overriding a patch or something.

**Comment by filoozom** [comment]
Okay, this one should be good to go. 👍

**Comment by filoozom** [comment]
Hey @ralphtheninja, have you had some time to test this out? If so, I could start working on moving all dependencies to git modules.

File: README.md [hunk 2]
```diff
@@ -294,6 +294,20 @@ There are multiple ways you can find help in using LevelDB in Node.js:
...
 
+### Git Submodules
+
+This project uses Git Submodules. This means that you should clone it recursively if you're planning on working on it:
+
+```bash
+$ git clone --recurse-submodules https://github.com/Level/leveldown.git
+```
+
+Alternatively, you can initalize submodules inside the cloned folder:
```
Comment (vweevers) [inline README.md]:
```suggestion
Alternatively, you can initialize submodules inside the cloned folder:
```

**Comment by ralphtheninja** [comment]
Nice!

**Pull request merged**